scripts/irods/test/test_itree.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import print_function
+
+import sys
+
+from . import session
+from .. import test
+
+if sys.version_info < (2,7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+
+class Test_ITree(session.make_sessions_mixin([('otherrods', 'rods')], []),
+                 unittest.TestCase):
+
+    def setUp(self):
+        super(Test_ITree, self).setUp()
+        self.admin = self.admin_sessions[0]
+
+    def tearDown(self):
+        super(Test_ITree, self).tearDown()
+
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_counting(self):
+        """Can Itree count objects properly?"""
+        self.admin.assert_icommand("itree", 'STDOUT_SINGLELINE', "Found 0 collections and 0 data objects")
+        for i in range(10):
+            self.admin.assert_icommand("imkdir {}".format(i))
```

Comment:
For idempotency, can we make the test collections as sub-collections of `self.admin.session_collection`? That collection gets cleaned up automatically on `tearDown()`. Maybe `setUp()` can `icd` to the session collection and then `icd` back to the original `ipwd` on `tearDown()` and that's all that would need to change

Otherwise, it looks like these will be left in the catalog after the test completes unless I'm missing something.

Reply:
I will add that to `setUp` then

Reply:
this appears to be done now...